sql/core/src/test/scala/org/apache/spark/sql/ScalaReflectionRelationSuite.scala

```diff
@@ -138,4 +144,16 @@ class ScalaReflectionRelationSuite extends SparkFunSuite with SharedSQLContext {
           Map(10 -> 100L, 20 -> 200L, 30 -> null),
           Row(null, "abc"))))
   }
+
+  test("decimal type with ScalaReflection") {
```

Comment:
Can we write the test in `ExpressionEncoderSuite`?  just add a line
`encodeDecodeTest(Decimal("32131413.211321313"), "catalyst decimal")`